### ISSUE TITLE
Better Package Organization

### ISSUE THREAD

**Issue by chewxy** [chewxy/gorgonia#91]
`Op` can be put into its own package
`Node` and `Graph` can be put into its own package
`Compile` can be put into its own package
`VM` can be put into its own package

**Comment by chewxy** [comment]
Experimental only. Culmination of rethinking that happened around #65

**Comment by pbberlin** [comment]
There is no tool to "break out" a type into a subpackage.
Alex Donovan started a tool "sock drawer" but it seems unfinished.
Once you have subpackages, you can move them around easily, using
`gomvpkg -from github.com/a/b/c/d  -to github.com/a/d`

If you encounter circular dependencies, after subpackaging,
you need to create further packages with the "joined" types.

Still I enjoyed the results of subpackaging my own project.

Regards 
  Peter

**Comment by chewxy** [comment]
Yeah... I was doing it by hand. There are a number of circular dependencies, and I am currently thinking about what the best move forwards is

**Comment by owulveryck** [comment]
Part of this issue is implemented with #307

**Comment by ArturTheAmazing** [comment]
Created a PR #481
